python/src/spark_rapids_ml/params.py

```diff
@@ -128,6 +128,36 @@ def _ensureIdCol(self, df: DataFrame) -> DataFrame:
         return df_withid
 
 
+class VerboseTypeConverters(TypeConverters):
+    @staticmethod
+    def _toIntOrBool(value: Any) -> Union[int, bool]:
+        if isinstance(value, bool):
+            return value
+
+        if TypeConverters._is_integer(value):
+            return int(value)
+
+        raise TypeError("Could not convert %s to Union[int, bool]" % value)
+
+
+class HasVerboseParam(Params):
+    """
+    Parameter to enable displaying verbose messages from cuml.
+    Refer to the cuML documentation for details on verbosity levels.
+    """
+
+    verbose: "Param[Union[int, bool]]" = Param(
```

Comment:
Why does verbose need to be a spark param?

Reply:
Same as above. copy only accepts Param type in the argument. Not sure if it is a good idea to change the function signature or mute the mypy type checking on the function.

Reply:
Ok.  For this and above let's make sure it the changes don't break anything.   copy() with extra params is only used in crossvalidator and changing verbose wouldn't be a part of that and we can ignore non-spark params for algo's in spark mllib.   
But if we're sure won't break anything, seems ok I guess to include these.

Otherwise, looks great.